### PR TITLE
add zulip integration

### PR DESCRIPTION
**Pull request by ziyangczi**
added zulip notification for our plugins addition like new plugin, new version, or removed plugin

### CODE REVIEW

File: backend/lambda.py [hunk 1]
```diff
@@ -383,6 +387,50 @@ def send_alert(message: str):
...
+            'to': 'hub-updates',
+            'topic': topic,
+            'content': message
+        }
+        response = requests.post('https://napari.zulipchat.com/api/v1/messages',
+                                 auth=HTTPBasicAuth(username, key), data=data)
+        if response.status_code != requests.codes.ok:
+            response.raise_for_status()
+    except HTTPError:
+        pass
```
Comment (codemonkey800) [inline backend/lambda.py:431]: Would be nice to have some handling for messages that fail to send. Maybe a queue that only gets dequeued if the request passes. It can be for another PR in case adding error handling for this is more difficult

**Review by codemonkey800 — APPROVED** [review]
Left one comment but overall LGTM 😄

File: backend/lambda.py [hunk 2]
```diff
@@ -383,6 +387,50 @@ def send_alert(message: str):
...
+
+    :param existing_packages: existing packages in cache
+    :param new_packages: new packages found
+    """
+    if zulip_credentials is not None:
+        username = zulip_credentials.split(":")[0]
+        key = zulip_credentials.split(":")[1]
+        for package, version in new_packages.items():
+            if package not in existing_packages:
+                send_zulip_message(username, key, package, f'New plugin published with version {version}!')
```
Comment (neuromusic) [inline backend/lambda.py:402]:
```suggestion
                send_zulip_message(username, key, package, f'A new plugin has been published on the napari hub! Check out [{package}](https://napari-hub.org/plugins/{package})!')
```

File: backend/lambda.py [hunk 3]
```diff
@@ -383,6 +387,50 @@ def send_alert(message: str):
...
+            elif existing_packages[package] != version:
+                send_zulip_message(username, key, package, f'A new version {version} has been published!')
```
Comment (neuromusic) [inline backend/lambda.py:404]:
```suggestion
                send_zulip_message(username, key, package, f'A new version of [{package}](https://napari-hub.org/plugins/{package}) is available on the napari hub! Check out [{version}](https://napari-hub.org/plugins/{package})!')
```

File: backend/lambda.py [hunk 4]
```diff
@@ -383,6 +387,50 @@ def send_alert(message: str):
...
+
+        for package, version in existing_packages.items():
+            if package not in new_packages:
+                send_zulip_message(username, key, package, f'This plugin has been unmarked as a plugin :(')
```
Comment (neuromusic) [inline backend/lambda.py:408]:
```suggestion
                send_zulip_message(username, key, package, f'This plugin is no longer available on the [napari hub](https://napari-hub.org) :(')
```

**Review by neuromusic — APPROVED** [review]
some suggested changes to the zulip messages. otherwise LGTM!

File: backend/lambda.py [hunk 5]
```diff
@@ -383,6 +387,50 @@ def send_alert(message: str):
...
+
+
+def send_zulip_message(username: str, key: str, topic: str, message: str):
```
Comment (justinelarsen) [inline backend/lambda.py:411]: Are there any limits on the rates of how many messages can be sent through the API in a certain time limit?

**Review by justinelarsen — APPROVED** [review]
LGTM, nice work!

File: backend/lambda.py (hunk 5 above)
Comment (ziyangczi) [reply backend/lambda.py:411]: from the reading I did there is none, hopefully we wouldn't send too many messages at once
